src/components/directory-content.js

```diff
@@ -821,7 +854,31 @@ const DirectoryContent = () => {
             })),
         [childrenMetadata, currentChildren, getElementTypeTranslation]
     );
+    const handleOpenDialog = useCallback(() => {
+        setOpenDialog(constants.DialogsId.ADD_ROOT_DIRECTORY);
+    }, [setOpenDialog]);
```

Comment:
since "setOpenDialog" is a "useState" setter, you don't need to put it in dependencies